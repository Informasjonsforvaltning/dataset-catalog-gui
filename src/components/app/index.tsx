import React, { FC } from "react";
import Main from "../pages/main";

const App: FC = () => <Main />;

export default App;
