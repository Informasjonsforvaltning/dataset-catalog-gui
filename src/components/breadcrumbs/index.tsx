import React, { FC, useEffect, useState } from 'react';
import { useParams, useLocation } from 'react-router-dom';

import { routes } from '../../router/routes';
import env from '../../utils/constants/env';
import { localization } from '../../utils/language/localization';
import SC from './styled';

const { CATALOG_PORTAL_BASE_URI } = env;

const Breadcrumbs: FC = () => {
  const [datasetId, setDatasetId] = useState('');

  const params = useParams();
  if (params.datasetId && params.datasetId !== datasetId) setDatasetId(params.datasetId);
  const activeStyle = { textDecoration: 'none' };

  let location = useLocation();
  useEffect(() => {
    setDatasetId('');
  }, [location]);

  const linkToDataset = datasetId ? `/${routes.home}/${params.datasetId}` : '';
  const linkToDatasets = `/${routes.home.replace(':catalogId', params.catalogId?.toString() ?? '')}`;

  return (
    <SC.BreadcrumbsNav>
      <span>
        <SC.ExternalLink aria-label={localization.allCatalogs} href={CATALOG_PORTAL_BASE_URI}>
          {localization.allCatalogs}
        </SC.ExternalLink>
        <span>
          <SC.CrumbDivider>{'>'}</SC.CrumbDivider>
          <SC.Link
            aria-label={localization.linkToDatasets}
            to={linkToDatasets}
            style={datasetId ? undefined : activeStyle}
            onClick={() => setDatasetId('')}
          >
            {localization.catalogType}
          </SC.Link>
        </span>
        {datasetId && (
          <span>
            <SC.CrumbDivider>{'>'}</SC.CrumbDivider>
            <SC.Link
              aria-label={localization.linkToDataset}
              to={linkToDataset}
              style={datasetId ? activeStyle : undefined}
            >
              {datasetId}
            </SC.Link>
          </span>
        )}
      </span>
    </SC.BreadcrumbsNav>
  );
};

export default Breadcrumbs;
