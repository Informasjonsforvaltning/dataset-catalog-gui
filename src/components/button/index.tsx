import React, { FC, ReactElement } from 'react';

import SC from './styled';

interface Props {
  label: string;
  style?: {};
  startIcon?: ReactElement;
  endIcon?: ReactElement;
  variant?: 'text' | 'contained' | 'outlined' | undefined;
}

const Button: FC<Props> = ({ label, startIcon, endIcon, style, variant = 'contained' }) => {
  return (
    <SC.Button startIcon={startIcon} endIcon={endIcon} variant={variant} sx={style}>
      {label}
    </SC.Button>
  );
};

export { Props };
export default Button;
