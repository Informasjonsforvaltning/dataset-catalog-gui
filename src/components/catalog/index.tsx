import React, { memo, FC } from 'react';
import { compose } from 'redux';
import memoize from 'lodash/memoize';
import { resolve } from 'react-resolver';

import { getDatasetsCount } from '../../services/api/registration-api/datasets';

import CatalogItem from '../catalog-item';

interface ExternalProps {
  catalogId: string;
  type: string;
  isReadOnly?: boolean;
  disabled: boolean;
}

interface Props extends ExternalProps {
  itemsCount?: number;
}

const Catalog: FC<Props> = ({
  catalogId,
  type,
  itemsCount,
  isReadOnly,
  disabled
}) => {
  const getLinkUri = () => `/catalogs/${catalogId}/${type}`;

  const linkUri = getLinkUri();

  return (
    <CatalogItem
      linkUri={linkUri}
      key={catalogId}
      type={type}
      itemsCount={itemsCount}
      isReadOnly={isReadOnly}
      disabled={disabled}
    />
  );
};

const memoizedGetDatasetsCount = memoize(getDatasetsCount);

const mapProps = {
  itemsCount: ({ type, catalogId, itemsCount }: any) => {
    switch (type) {
      case 'datasets': {
        return memoizedGetDatasetsCount(catalogId);
      }
      default:
        return itemsCount;
    }
  }
};

export default compose<FC<ExternalProps>>(memo, resolve(mapProps))(Catalog);
