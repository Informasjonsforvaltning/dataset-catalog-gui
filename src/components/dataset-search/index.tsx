import { Colour, theme } from '@fellesdatakatalog/theme';
import React, { FC } from 'react';
import { ACTION_TYPE } from '../../context/actions';
import { useTableDispatch } from '../../context/table-context';
import { localization } from '../../utils/language/localization';
import DropDown from '../inputs/dropdown';
import InputField from '../inputs/input-field';
import SC from './styled';

const Search: FC = () => {
  const options: [string, string][] = Object.entries(localization.tag);

  const tableDispatch = useTableDispatch();

  const onInputSubmit = (searchTerm: string) =>
    tableDispatch({ type: ACTION_TYPE.FILTER_DATASETS, payload: { type: 'search', value: searchTerm } });

  const onDropdownSelect = (status: string) =>
    tableDispatch({
      type: ACTION_TYPE.FILTER_DATASETS,
      payload: {
        type: 'status',
        value: getEnglishStatus(status),
      },
    });

  const getBokmaalStatus = (): (string | undefined)[] =>
    options
      .map(status => {
        return status ? localization.tag[status[0].toLowerCase()] : ' ';
      })
      .filter(status => status);

  const getEnglishStatus = (status: string): string => {
    const engStatus = options.find(option => (option[1].toLowerCase() === status.toLowerCase() ? option[0] : ''));
    return engStatus ? engStatus[0] : '';
  };

  return (
    <SC.Search>
      <InputField ariaLabel={localization.searchForDataset} onInputSubmit={onInputSubmit} />
      <DropDown
        name={localization.dropdown.statusSearch}
        bg={theme.colour(Colour.BLUE, 'B30')}
        dropdownColor={theme.colour(Colour.BLUE, 'B60')}
        options={getBokmaalStatus()}
        onDropdownSelect={onDropdownSelect}
      />
    </SC.Search>
  );
};

export default Search;
