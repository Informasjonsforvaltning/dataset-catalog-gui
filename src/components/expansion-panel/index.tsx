import React, { FC, PropsWithChildren, useState, useId } from "react";
import SC from './styled';

import Header from './header';

interface Props {
  headerTitle: string;
  tag?: 'mandatory' | 'recommended';
  hasError: boolean;
  summary: string;
  initiallyExpanded: boolean;
}

const ExpansionPanel: FC<PropsWithChildren<Props>> = ({
  headerTitle,
  tag,
  hasError = false,
  initiallyExpanded = false,
  summary,
  children
}) => {

  const [isExpanded, setIsExpanded] = useState(initiallyExpanded);

  const toggleExpansion = () => {
    setIsExpanded(!isExpanded);
  }

  const handleClick = () => {
    toggleExpansion();
  }

  const headerId = useId();
  const panelId = useId();
  
  return (
      <SC.ExpansionPanel
        isExpanded={isExpanded}
        aria-labelledBy={headerId}>
        <Header
          headerTitle={headerTitle}
          tag={tag}
          hasError={hasError}
          summary={summary}
          isExpanded={isExpanded}
          headerId={headerId}
          controlledPanelId={panelId}
          onClick={handleClick}
        />
        <>
          {isExpanded 
            ? <SC.ChildWrapper
              id={panelId}
              isExpanded={isExpanded}
              >
                {children}
              </SC.ChildWrapper>
            : <></> }
        </>
      </SC.ExpansionPanel>
  )
}

export default ExpansionPanel;
