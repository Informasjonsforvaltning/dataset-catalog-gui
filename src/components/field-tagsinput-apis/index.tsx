import React, { memo, FC, useState, useEffect } from 'react';
import { compose } from 'redux';
import ReactTags, { Tag } from 'react-tag-autocomplete';
import type { WrappedFieldProps } from 'redux-form';

import {
  withTranslations,
  Props as TranslationsProps
} from '../../providers/translations';

import {
  extractSuggestions,
  getDataserviceSuggestions
} from '../../services/api/fulltext-search/suggestions';

interface Props extends WrappedFieldProps, TranslationsProps {}

const InputTagsAPIsField: FC<Props> = ({
  input: { value, onChange },
  translationsService
}) => {
  const [tags, setTags] = useState<Tag[]>([]);
  const [suggestions, setSuggestions] = useState<Tag[]>([]);

  const addTagToInput = (updates: any) =>
    onChange(
      (value || []).concat({
        id: updates.id,
        description: {
          [translationsService.getLanguage()]: updates.name
        },
        endpointDescription: [
          {
            uri: updates.id
          }
        ]
      })
    );

  const removeTagFromInput = (index: number) => {
    value.splice(index, 1);
    onChange(value);
  };

  const loadSuggestions = (q: string) =>
    getDataserviceSuggestions({ q })
      .then(extractSuggestions)
      .then(items =>
        setSuggestions(
          items.map(({ id, title }: any) => ({
            id,
            name: translationsService.translate(title)
          }))
        )
      )
      .catch(() => {});

  const handleDelete = (i: number) => {
    const newTags = [...tags];
    newTags.splice(i, 1);
    setTags(newTags);
    removeTagFromInput(i);
  };

  const handleAddition = (tag: Tag) => {
    setTags([...tags, tag]);
    addTagToInput(tag);
  };

  useEffect(() => {
    if (value) {
      const accessServices = Array.isArray(value) ? value : [value];

      setTags(
        accessServices.filter(Boolean).map(({ id, description }) => ({
          id,
          name: translationsService.translate(description)
        }))
      );
    }
  }, []);

  return (
    <div className='pl-2'>
      <div className='d-flex align-items-center'>
        <ReactTags
          autofocus={false}
          placeholderText=''
          tags={tags}
          minQueryLength={1}
          suggestions={suggestions}
          onDelete={handleDelete}
          onAddition={handleAddition}
          onInput={loadSuggestions}
        />
      </div>
    </div>
  );
};

export default compose<FC>(memo, withTranslations)(InputTagsAPIsField);
