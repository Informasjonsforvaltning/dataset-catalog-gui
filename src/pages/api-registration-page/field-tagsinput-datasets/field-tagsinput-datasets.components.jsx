import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import ReactTags from 'react-tag-autocomplete';
import { getTranslateText } from '../../../services/translateText';
import { searchDatasets } from '../../../services/api/search-api/datasets';

const addTagToInput = (updates, props) => {
  const { input } = props;
  let inputValues = input.value;
  if (!inputValues) {
    inputValues = [];
  }
  inputValues.push(_.get(updates, ['id']));
  input.onChange(inputValues);
};

const removeTagFromInput = (index, props) => {
  const { input } = props;
  const inputValues = input.value;

  inputValues.splice(index, 1);
  input.onChange(inputValues);
};

export class InputTagsDatasetsField extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tags: [],
      suggestions: []
    };
    this.loadSuggestions = this.loadSuggestions.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
    this.handleAddition = this.handleAddition.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentDidMount() {
    const { input, referencedDatasets } = this.props;
    const datasetReferences = [];
    _.get(input, 'value', []).forEach(item => {
      datasetReferences.push({
        id: item,
        name: getTranslateText(
          _.get(_.find(referencedDatasets, ['uri', item]), 'title')
        )
      });
    });
    this.setState({
      tags: datasetReferences
    });
  }

  handleDelete(i) {
    const { tags } = this.state;
    const newTags = [...tags];
    newTags.splice(i, 1);
    this.setState({ tags: newTags });
    removeTagFromInput(i, this.props);
  }

  handleAddition(tag) {
    const { tags } = this.state;
    this.setState({ tags: [...tags, tag] });
    addTagToInput(tag, this.props);
  }

  handleInputChange(input) {
    this.loadSuggestions(input);
  }

  loadSuggestions(value) {
    const datasets = [];

    searchDatasets({
      title: value,
      returnFields: 'title,uri'
    })
      .then(responseData => {
        _.get(responseData, ['hits', 'hits'], []).forEach(item => {
          datasets.push({
            id: _.get(item, ['_source', 'uri']),
            name: _.get(item, ['_source', 'title', 'nb'])
          });
        });
        this.setState({
          suggestions: datasets
        });
      })
      .catch(error => {
        if (process.env.NODE_ENV !== 'production') {
          console.log('error', error); // eslint-disable-line no-console
        }
      });
  }

  render() {
    const { tags, suggestions } = this.state;
    return (
      <div className="pl-2">
        <div className="d-flex align-items-center">
          <ReactTags
            autofocus={false}
            placeholder=""
            tags={tags}
            minQueryLength={1}
            suggestions={suggestions}
            handleDelete={this.handleDelete}
            handleAddition={this.handleAddition}
            handleInputChange={this.handleInputChange}
          />
        </div>
      </div>
    );
  }
}

InputTagsDatasetsField.defaultProps = {
  input: null,
  referencedDatasets: null
};

InputTagsDatasetsField.propTypes = {
  input: PropTypes.object,
  referencedDatasets: PropTypes.array
};
