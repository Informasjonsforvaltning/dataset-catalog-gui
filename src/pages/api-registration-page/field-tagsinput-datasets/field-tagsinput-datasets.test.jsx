import React from 'react';
import { shallow } from 'enzyme';
import { InputTagsDatasetsField } from './field-tagsinput-datasets.components';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    input: {
      name: 'datasetReferences',
      value: [
        'http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12710',
        'http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12711'
      ]
    },
    referencedDatasets: [
      {
        id: 'efe0948d-c029-4e61-a99a-b0a6fcb420ae',
        uri:
          'http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12710',
        title: {
          nb: 'Tittel datasett'
        }
      },
      {
        id: 'efe0948d-c029-4e61-a99a-b0a6fcb420ae',
        uri:
          'http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12711',
        title: {
          nb: 'Tittel datasett 1'
        }
      }
    ]
  };
  wrapper = shallow(<InputTagsDatasetsField {...defaultProps} />);
});

test('should render InputTagsDatasetsField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
