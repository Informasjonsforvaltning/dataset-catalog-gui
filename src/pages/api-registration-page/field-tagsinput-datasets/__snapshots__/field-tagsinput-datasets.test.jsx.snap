// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputTagsDatasetsField correctly 1`] = `
<div
  className="pl-2"
>
  <div
    className="d-flex align-items-center"
  >
    <ReactTags
      addOnBlur={false}
      allowBackspace={true}
      allowNew={false}
      autofocus={false}
      autoresize={true}
      clearInputOnDelete={true}
      delimiterChars={Array []}
      delimiters={
        Array [
          9,
          13,
        ]
      }
      handleAddition={[Function]}
      handleDelete={[Function]}
      handleInputChange={[Function]}
      inputAttributes={Object {}}
      maxSuggestionsLength={6}
      minQueryLength={1}
      placeholder=""
      suggestions={Array []}
      suggestionsFilter={null}
      tagComponent={null}
      tags={
        Array [
          Object {
            "id": "http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12710",
            "name": "Tittel datasett",
          },
          Object {
            "id": "http://brreg.no/catalogs/910244132/datasets/6deb833c-4600-4377-8f9b-db0aacf12711",
            "name": "Tittel datasett 1",
          },
        ]
      }
    />
  </div>
</div>
`;
